tests/usbredir_test.go

```diff
@@ -98,25 +103,26 @@ var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-c
 		})
 
 		It("Should fail when limit is reached", func() {
-			cancelFns := make([]context.CancelFunc, v1.UsbClientPassthroughMaxNumberOf+1)
-			errors := make([]chan error, v1.UsbClientPassthroughMaxNumberOf+1)
+			var tests [v1.UsbClientPassthroughMaxNumberOf + 1]testCase
```

Comment:
nit: You can just declare the time and not allocate all the structs, `append` works just fine, no need to pre-alloc...

Reply:
I forgot do address this...  You can consider it a C mentality...  We know beforehand the amount of memory we need, it is local to the test and as such an array seems reasonable wrt memory efficiency. I'll change it to slice as is more idiomatic in Go

Reply:
Added this

```diff
--- a/tests/usbredir_test.go
+++ b/tests/usbredir_test.go
@@ -107,29 +107,31 @@ var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-c
                })

                It("Should fail when limit is reached", func() {
-                       var tests [v1.UsbClientPassthroughMaxNumberOf + 1]testCase
+                       var tests []testCase
                        for i := 0; i <= v1.UsbClientPassthroughMaxNumberOf; i++ {
                        retry_loop:
                                for try := 0; try < 3; try++ {
                                        ctx, cancelFn := context.WithCancel(context.Background())
-                                       tests[i] = testCase{
+                                       test := testCase{
                                                cancel:  cancelFn,
                                                connect: make(chan struct{}),
                                                err:     make(chan error),
                                        }
-                                       ctx = context.WithValue(ctx, "connected", tests[i].connect)
-                                       go runConnectGoroutine(virtClient, name, namespace, ctx, tests[i].err)
+                                       ctx = context.WithValue(ctx, "connected", test.connect)
+                                       go runConnectGoroutine(virtClient, name, namespace, ctx, test.err)

                                        if i == v1.UsbClientPassthroughMaxNumberOf {
                                                // Last test is meant to fail.
+                                               tests = append(tests, test)
                                                break
                                        }

                                        // Till the last test, all sockets must be connected
                                        select {
-                                       case <-tests[i].connect:
+                                       case <-test.connect:
+                                               tests = append(tests, test)
                                                break retry_loop
-                                       case err := <-tests[i].err:
+                                       case err := <-test.err:
                                                if !errors.Is(err, syscall.ECONNRESET) {
                                                        log.Log.Reason(err).Info("Failed early. Unexpected error.")
                                                        Fail("Improve error handling or fix underlying issue")
@@ -137,7 +139,7 @@ var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-c
                                                log.Log.Reason(err).Infof("Failed early. Try again (%d)", try)
                                        case <-time.After(time.Second):
                                                log.Log.Infof("Took too long. Try again (%d)", try)
-                                               tests[i].cancel()
+                                               test.cancel()
```